runner/config.ts

```diff
@@ -285,11 +290,11 @@ interface PreparsedArgs {
  * @param {string} root
  * @return {!Object} The merged configuration.
  */
-export function fromDisk(jsonOnly?: boolean, root?: string): Config {
+export function fromDisk(jsonOnly?: boolean, root?: string, configFile?: string): Config {
   const matcher = jsonOnly ? JSON_MATCHER : CONFIG_MATCHER;
 
   const globalFile = path.join(HOME_DIR, matcher);
-  const projectFile = findup(matcher, {nocase: true, cwd: root});
+  const projectFile = findup(configFile || matcher, {nocase: true, cwd: root});
```

Comment:
Can we handle this in runner/context.ts instead? Instead of overloading fromDisk, lets change it's function signature to: 

```js
fromDisk(matcher: string, root?: string): Config {
```

and then in context.ts:


```js
let matcher;
if (configFile) {
  matcher = configFile;
} else if (jsonOnly) {
  matcher = JSON_MATCHER;
} else {
  matcher = CONFIG_MATCHER;
}
```